xbmc/windowing/wayland/GLContextEGL.cpp

```diff
@@ -0,0 +1,199 @@
+/*
+ *      Copyright (C) 2017 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "GLContextEGL.h"
+
+#include "utils/EGLUtils.h"
+#include "utils/log.h"
+
+using namespace KODI::WINDOWING::WAYLAND;
+
+CGLContextEGL::CGLContextEGL()
+{
+  // EGL_EXT_platform_wayland requires EGL_EXT_client_extensions, so this should
+  // be safe to use
+  m_clientExtensions = CEGLUtils::GetClientExtensions();
+
+  if (m_clientExtensions.find("EGL_EXT_platform_base") == m_clientExtensions.end())
+  {
+    throw std::runtime_error("EGL implementation does not support EGL_EXT_platform_base, cannot continue");
+  }
+  if (m_clientExtensions.find("EGL_EXT_platform_wayland") == m_clientExtensions.end())
+  {
+    throw std::runtime_error("EGL implementation does not support EGL_EXT_platform_wayland, cannot continue");
+  }
+
+  m_eglGetPlatformDisplayEXT = CEGLUtils::GetRequiredProcAddress<PFNEGLGETPLATFORMDISPLAYEXTPROC>("eglGetPlatformDisplayEXT");
+  m_eglCreatePlatformWindowSurfaceEXT = CEGLUtils::GetRequiredProcAddress<PFNEGLCREATEPLATFORMWINDOWSURFACEEXTPROC>("eglCreatePlatformWindowSurfaceEXT");
+}
+
+CGLContextEGL::~CGLContextEGL()
+{
+  Destroy();
+}
+
+bool CGLContextEGL::CreateDisplay(wayland::display_t& display,
+                                  EGLint renderable_type,
+                                  EGLint rendering_api)
+{
+  EGLint neglconfigs = 0;
+  int major, minor;
+
+  EGLint attribs[] ={
```

Comment:
Depth buffer added - music visualization was indeed looking funny without it :-)
My understanding is that we would need alpha buffer on the surface only if the Kodi window should have transparency, which I think is not the case